src/main/java/de/iav/mucjava231burgershop/repository/MenuRepository.java

```diff
@@ -0,0 +1,66 @@
+package de.iav.mucjava231burgershop.repository;
+
+import de.iav.mucjava231burgershop.exception.MenuNotFoundException;
+import de.iav.mucjava231burgershop.model.Menu;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.ArrayList;
+import java.util.List;
+@Repository
+public class MenuRepository {
+
+    private final List<Menu> menus;
+
+    public MenuRepository(){
+        menus = new ArrayList<>();
+    }
+    public MenuRepository(List<Menu> menus) {
+        this.menus = menus;
+    }
+
+    public List<Menu> menuList(){
+        return menus;
+    }
+
+    public Menu getMenuById(String idOfRequestedMenu){
+        for (Menu singleMenuFromList : menus){
+            if (singleMenuFromList.id().equals(idOfRequestedMenu)) {
+                return singleMenuFromList;
+            }
+        }
+        throw new MenuNotFoundException(idOfRequestedMenu);
+    }
+
+    public void addMenu(Menu menuToAdd){
+        menus.add(menuToAdd);
+    }
+
+    public Menu changeMenu(String id, Menu menuToChange) {
+
+        for (Menu singleMenuFromList : menus) {
+            if (singleMenuFromList.id().equals(id)) {
+                menus.remove(singleMenuFromList);
+                Menu changedMenu = new Menu(id,
+                        menuToChange.name(),
+                        menuToChange.price(),
+                        menuToChange.mainDish(),
+                        menuToChange.sideDish(),
+                        menuToChange.beverage());
+                        menus.add(changedMenu);
```

Comment:
Ganz einfach neue Objekte mit einem veränderten Attribut erstellen geht super mit einer `with`-Hilfsmethode
Finde im Internet grad nicht die Quelle, deshalb von ChatGPT einmal das Beispiel erstellt was ich meine:

---
"Give an example how to use the "with" Syntax to create new immutable objects with Java Records"

Certainly! In Java, you can use the with syntax with records to create new immutable objects with modified values. The with syntax allows you to create a new instance of a record by copying the values of an existing instance and modifying specific fields. Here's an example:

```java
public record Person(String name, int age) {
    public Person withAge(int newAge) {
        return new Person(name, newAge);
    }
}

public class Example {
    public static void main(String[] args) {
        Person person1 = new Person("Alice", 25);
        System.out.println("Original Person: " + person1);

        Person person2 = person1.withAge(30);
        System.out.println("Modified Person: " + person2);
    }
}
```

When you run this code, it will output the same result as before:

```
Original Person: Person[name=Alice, age=25]
Modified Person: Person[name=Alice, age=30]
```

In this example, we have a record called Person with two fields: name and age. The with syntax allows us to create a new Person instance (person2) by modifying the age field of an existing instance (person1). The withAge method is automatically generated by the record and takes the new age value as a parameter.